test/unittest/tests/block/test_block_api_misc.cpp

```diff
@@ -0,0 +1,242 @@
+}
+
+static void
+check_bm_stats(WT_SESSION_IMPL *session, WT_BM *bm)
+{
+    WT_DSRC_STATS stats;
+
+    // Enable statistics on the connection to allow the statistic macros to update correctly.
+    S2C(session)->stat_flags = 1;
+    REQUIRE(bm->stat(bm, session, &stats) == 0);
+    CHECK(stats.allocation_size == bm->block->allocsize);
+    CHECK(stats.block_checkpoint_size == (int64_t)bm->block->live.ckpt_size);
+    CHECK(stats.block_magic == WT_BLOCK_MAGIC);
+    CHECK(stats.block_major == WT_BLOCK_MAJOR_VERSION);
+    CHECK(stats.block_minor == WT_BLOCK_MINOR_VERSION);
+    CHECK(stats.block_reuse_bytes == (int64_t)bm->block->live.avail.bytes);
+    CHECK(stats.block_size == bm->block->size);
+
+    // Disable statistics on the connection when finished so that the mock session destructor
+    // doesn't try to dereference invalid memory.
+    S2C(session)->stat_flags = 0;
+}
+
+// Test that the block manager's addr_string method produces the expected string representation.
+static void
+test_addr_string(WT_SESSION_IMPL *session, WT_BM *bm, wt_off_t pack_offset, uint32_t pack_size,
+  uint32_t pack_checksum, std::string expected_str)
+{
+    // Initialize a buffer.
+    WT_ITEM buf;
+    WT_CLEAR(buf);
+
+    // Generate an address cookie - technically, we shouldn't know about internal details of the
+    // address cookie, but this allows for more rigorous testing with different inputs.
+    uint8_t p[WT_BTREE_MAX_ADDR_COOKIE], *pp;
+    pp = p;
+    REQUIRE(__wt_block_addr_pack(
+              bm->block, &pp, WT_TIERED_OBJECTID_NONE, pack_offset, pack_size, pack_checksum) == 0);
+    size_t addr_size = WT_PTRDIFF(pp, p);
+
+    // Compare the string output of bm->addr_string against the known expected string.
+    pp = p;
+    REQUIRE(bm->addr_string(bm, nullptr, &buf, pp, addr_size) == 0);
+    CHECK(static_cast<std::string>(((char *)(buf.data))).compare(expected_str) == 0);
+
+    __wt_free(session, buf.data);
+}
+
+// Tested using a white-box approach as we need knowledge of internal structures to test various
+// inputs.
+TEST_CASE("Block manager addr string", "[block_api_misc]")
+{
+    std::shared_ptr<mock_session> session;
+
+    WT_BM bm;
+    WT_CLEAR(bm);
+    initialize_bm(session, &bm);
+    WT_SESSION_IMPL *s = session->get_wt_session_impl();
+
+    SECTION("Test addr string with non-zero values")
+    {
+        // (512, 1024, 12345) -> (offset, size, checksum)
+        test_addr_string(s, &bm, 512, 1024, 12345, "[0: 512-1536, 1024, 12345]");
+    }
+
+    SECTION("Test addr string with zero values")
+    {
+        // (0, 0, 0) -> (offset, size, checksum)
+        test_addr_string(s, &bm, 0, 0, 0, "[0: 0-0, 0, 0]");
+    }
+
+    SECTION("Test addr string with zero size")
+    {
+        // (512, 0, 12345) -> (offset, size, checksum)
+        test_addr_string(s, &bm, 512, 0, 12345, "[0: 0-0, 0, 0]");
+    }
+
+    REQUIRE(__wt_block_close(s, bm.block) == 0);
+}
+
+// Tested using a black-box approach.
```

Comment:
I don't think we need to comment which style we are going for. It would create more confusion than not. I'm on the side where we collect everything under unit testing instead.